sirgalleto/chapter-1/first-problem.js

```diff
@@ -0,0 +1,39 @@
+const assert = require('../helpers/assert');
+
+function isUniqueCharacterString(string) {
+  const characterMap = {};
+
+  for (const character of string) {
+    if(characterMap[character]) {
+      return false
+    } else {
+      characterMap[character] = true;
+    }
+  }
+  return true;
+}
+
+function isUniqueCharacterStringNoDataStructure(string) {
+  for (let index = 0; index < string.length; index++) {
+    for (let secondaryIndex = index + 1; secondaryIndex < string.length - 1; secondaryIndex++) {
+      if (string[index] === string[secondaryIndex]) {
```

Comment:
this code does not work as intended. this only checks the case in which the letter being evaluated is equal to the previous one.